### PR TITLE
 [env] 🐛  define provider per common environment instance

### PR DESCRIPTION
**Pull request by pducolin**
What does this PR do?
---------------------

Define one providers per environment instance

Which scenarios this will impact?
-------------------

All

Motivation
----------

Default singleton providers are causing a bug when updating the environment, as pulumi keeps re-using the same instance of provider while creating a new environment instance

Additional Notes
----------------

### CODE REVIEW

**Review by usamasaqib — APPROVED** [review]
Thanks!

File: aws/scenarios/fakeintake/run.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package fakeintake
+
+import (
```
Comment (ogaca-dd) [inline aws/scenarios/fakeintake/run.go:3]: Note: This file doesn't seem to belong to this PR

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: aws/scenarios/fakeintake/run.go (hunk 1 above)
Comment (pducolin) [reply aws/scenarios/fakeintake/run.go:3]: ah good catch, thank you !

**Review by vboulineau — APPROVED** [review]

**Comment by pducolin** [comment]
@ogaca-dd I forced push a reset of the history, removing the fakeintake run that was not supposed to make it to this PR
